### PR TITLE
Fix #3705 - use GENS case_name option

### LINKED ISSUE #3705: New GENS case name key for queries
Add a `case_name` with the case name to complement the earlier sample display name as a query GET parameter when linking out to GENS (case and SV variant page). Is display name or internal name intended, @raysloks? I'm guessing the easier way is internal id (pet name), as this is unique across all institutes.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Could be tested with latest GENS on stage. Or just check in output that `case_name` is added to the two places where we link out to GENS.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: scout/server/blueprints/variant/templates/variant/sv-variant.html [hunk 1]
```diff
@@ -221,7 +221,7 @@
           <span class="d-inline">Copy number profile</span>
           <span class="d-inline ms-3">
             {% for sample in variant.samples %}
-              <a class="btn btn-secondary" target="_blank" href="http://{{gens_info.host}}/{{sample.display_name}}?region={{variant.chromosome}}:{{variant.position - 2000}}-{{variant.end + 2000}}&variant={{variant._id}}&genome_build={{gens_info.genome_build}}", {{variant.chromosome}})>Gens {{ sample.display_name }}</a>
+              <a class="btn btn-secondary" target="_blank" href="http://{{gens_info.host}}/{{sample.display_name}}?region={{variant.chromosome}}:{{variant.position - 2000}}-{{variant.end + 2000}}&variant={{variant._id}}&genome_build={{gens_info.genome_build}}&case_name={{case.case_id}}", {{variant.chromosome}})>Gens {{ sample.display_name }}</a>
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/sv-variant.html:224]: Will this change have an effect also on other gens instances different from ours?

File: scout/server/blueprints/cases/templates/cases/individuals_table.html [hunk 2]
```diff
@@ -126,7 +126,7 @@
                   </div>
                 </label>
                 {% if gens_info.display %}
-                  <a style="font-size: .8rem;" class="btn btn-secondary btn-sm" target="_blank" href="http://{{gens_info.host}}/{{ind.display_name}}?&genome_build={{case.genome_build}}">CN profile</a>
+                  <a style="font-size: .8rem;" class="btn btn-secondary btn-sm" target="_blank" href="http://{{gens_info.host}}/{{ind.display_name}}?&genome_build={{case.genome_build}}&case_id={{case.case_id}}">CN profile</a>
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/templates/cases/individuals_table.html:129]: Code-wise it looks good. Have you tried it on stage?

Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/individuals_table.html:129]: Not yet! Our local fork still uses `case_name`, but @raysloks relevantly suggested switching to `case_id` . Are you on that one, @raysloks, or should we PR it? 😊

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -333,6 +333,9 @@ def case(store, institute_obj, case_obj):
     # Limit secondary findings according to institute settings
     limit_genes = store.safe_genes_filter(institute_obj["_id"])
 
+    # Make sure build is either "37" or "38"
+    build = "38" if "38" in str(case_obj["genome_build"]) else "37"
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:337]:
```suggestion
    build = "38" if "38" in str(case_obj["genome_build"]) else "37"
```
```suggestion
    build = "38" if "38" in str(case_obj.get("genome_build")) else "37"
```

File: scout/server/blueprints/cases/templates/cases/individuals_table.html (hunk 2 above)
Comment (raysloks) [reply scout/server/blueprints/cases/templates/cases/individuals_table.html:129]: Yeah, I'm going to be pushing the swap to `case_id` on Gens quite soon.

File: scout/server/blueprints/cases/controllers.py (hunk 3 above)
Comment (dnil) [reply scout/server/blueprints/cases/controllers.py]: I mean, it really should exist, but sure.

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py]: It doesn't for some very old cases. I remember this caused a bug a few years ago

**Comment by dnil** [comment]
And now link work to the updated gens works! Nice @raysloks!

Screenshot 2022-12-14 at 12 44 39

**Review by northwestwitch — APPROVED** [review]
Looks good. Haven't tried but your test worked, so it's fine to merge I think! 👍🏻

**Pull request merged**